src/rust/engine/process_execution/src/lib.rs

```diff
@@ -396,7 +399,10 @@ impl CommandRunner for BoundedCommandRunner {
     self
       .inner
       .1
-      .with_acquired(move || inner.0.run(req, context))
+      .clone()
+      .with_acquired(move || inner.0.run(req, context).compat())
```

Comment:
Just to check that I understand this slightly fiddly method chain:
* The first `.compat()` here is to convert the futures-crate Future into a std Future, as required by the `with_acquired` signature. This will go away when `CommandRunner` is updated to return a std Future.
* `.boxed()` here is to make the returned future `Pin` which is required so that it can be converted back into a futures-crate Future. The compat after, and _probably_ this, can disappear after `CommandRunner` is updated to return std Future.
* The second `.compat()` then converts back into a futures-crate Future.
* `.to_boxed()` is required because the trait returns a `BoxFuture` - at some point we'll update this method (and `boxfuture` in general) to return a `Box<std::future::Future>` instead of a `Box<futures::Future>`, but we'll still need the `.to_boxed()` for the foreseeable future because the underlying reason it's necessary haven't changed.

Is that right?

Reply:
That is correct. The only thing that feels truly extraneous is that `.boxed()` causes the 3->1 `.compat()` call to fail, unfortunately.